k8s/k8s.go

```diff
@@ -2665,76 +2696,143 @@ func (k *k8sOps) ListEvents(namespace string, opts meta_v1.ListOptions) (*v1.Eve
 	return k.client.CoreV1().Events(namespace).List(opts)
 }
 
+func (k *k8sOps) CreateCRD(resources []CustomResource) error {
+	for _, resource := range resources {
+		err := k.createCRD(resource)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, resource := range resources {
+		if err := k.waitForCRDInit(resource); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (k *k8sOps) createCRD(resource CustomResource) error {
+	crdName := fmt.Sprintf("%s.%s", resource.Plural, resource.Group)
+	crd := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: crdName,
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   resource.Group,
+			Version: resource.Version,
+			Scope:   resource.Scope,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Singular: resource.Name,
+				Plural:   resource.Plural,
+				Kind:     resource.Kind,
+			},
+		},
+	}
+
+	_, err := k.apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
```

Comment:
I feel we should delegate the error handling to the caller. Sometimes it is better to know it is already created.

Reply:
updated